<launch>
  <arg name="robot_name" default="marrwy" />
  <arg name="map_name" default="map" />
  <arg name="base_frame" default="base_frame" />
  <arg name="laser_topic" default="/scan" />

  <arg name="initial_pose_x" default="0.0" />
  <arg name="initial_pose_y" default="0.0" />
  <arg name="initial_pose_a" default="0.0" />

  <arg name="paramsdir" default="$(find marrtino_r3d)/config/$(arg robot_name)" />
  <arg name="mapsdir" default="$(find marrtino_r3d)/maps" />

  <!-- Static transforms in the system 
  <node pkg="marrtino_r3d" type="static_broadcaster.py" name="static_broadcaster_node"/>
  -->

  <node pkg="map_server" type="map_server" name="map_server" args="$(arg mapsdir)/$(arg map_name).yaml" respawn="false" >
        <param name="frame_id" value="map" />
   </node>
  <!-- amcl -->
  <node pkg="amcl" type="amcl" name="amcl" respawn="true">
  <!--
    <remap from="scan" to="$(arg laser_topic)"/>
  -->
    <!-- Default parameters   -->
    <rosparam file="$(find marrtino_r3d)/config/$(arg robot_name)/amcl.yaml" command="load" />

   <!--<rosparam file="$(find marrtino_r3d)/config/rodney/amcl_config.yaml" command="load" />
    -->
    <!-- Override Params to include prefix -->
    <param name="global_frame_id" value="/map" /> 
    <param name="base_frame_id" value="/$(arg base_frame)" />
    <param name="odom_frame_id" value="/odom" />

    <!-- Set initial pose -->
    <param name="initial_pose_x" value="$(arg initial_pose_x)" />
    <param name="initial_pose_y" value="$(arg initial_pose_y)" />
    <param name="initial_pose_a" value="$(arg initial_pose_a)" />
    <param name="initial_cov_xx" value="0.25" />
    <param name="initial_cov_yy" value="0.25" />
    <param name="initial_cov_aa" value="0.25" />

  </node>
  <!--- Run Move Base --> 
  <include file="$(find marrtino_r3d)/launch/move_base.launch"/>

   
 </launch>
